### PR TITLE
Do not run TryParseExact on null dates

### PR DESCRIPTION
**Pull request by pakrym**
# Before

# After

@davidfowl @Tratcher

### CODE REVIEW

File: src/Microsoft.Net.Http.Headers/HttpRuleParser.cs [hunk 1]
```diff
@@ -241,6 +241,11 @@ internal static string DateToString(DateTimeOffset dateTime)
 
         internal static bool TryStringToDate(string input, out DateTimeOffset result)
         {
+            if (input == null)
```
Comment (pranavkm) [inline src/Microsoft.Net.Http.Headers/HttpRuleParser.cs]: Does this help to add a check for empty strings or is that not a common occurence?

Comment (pakrym) [inline src/Microsoft.Net.Http.Headers/HttpRuleParser.cs]: It usually not. This is for the case when header is absent.

**Comment by Tratcher** [comment]
:shipit:

**Comment by pakrym** [comment]
#  After null short-circuit
!(http://i.imgur.com/ADALzcc.png)[http://i.imgur.com/ADALzcc.png]

**Comment by davidfowl** [comment]
🚢 🇮🇹

**Pull request merged**